Makefile

```diff
@@ -50,109 +51,127 @@ export CGO_ENABLED=0
 
 BINDIR:=/usr/bin/
 
-# this target must be listed first in order for it to be a default target,
-# so that ubuntu_ppa's may be constructed using default build tools.
+# .DEFAULT_GOAL := build
+# .PHONY: clean build format imports lint test unit acceptance prepare-for-pr verify verify-format benchmark 
+
+## bUild: Build the program
 build: out
-	@echo "> Building..."
+	@echo "=====> Building..."
 	$(GOCMD) build -ldflags "-s -w -X 'github.com/buildpacks/pack.Version=${PACK_VERSION}' -extldflags ${LDFLAGS}" -trimpath -o ./out/$(PACK_BIN) -a ./cmd/pack
 
+## all: Run clean, verify, test, and build operations
 all: clean verify test build
 
-# used by apt-get install when installing ubuntu ppa.
-# move pack binary onto a path location.
-install:
-	mkdir -p ${DESTDIR}${BINDIR}
-	cp ./out/$(PACK_BIN) ${DESTDIR}${BINDIR}/
-
-mod-tidy:
-	$(GOCMD) mod tidy  -compat=1.19
-	cd tools && $(GOCMD) mod tidy -compat=1.19
-
-tidy: mod-tidy format
-
-package: out
-	tar czf .$/out$/$(ARCHIVE_NAME).tgz -C .$/out$/ $(PACK_BIN)
-
-install-mockgen:
-	@echo "> Installing mockgen..."
-	cd tools && $(GOCMD) install github.com/golang/mock/mockgen
-
-install-goimports:
-	@echo "> Installing goimports..."
-	cd tools && $(GOCMD) install golang.org/x/tools/cmd/goimports
+## clean: Clean the workspace
+clean:
+	@echo "=====> Cleaning workspace..."
+	@$(RMRF) .$/out benchmarks.test || (exit 0)
 
+## format: Format the code
 format: install-goimports
-	@echo "> Formating code..."
+	@echo "=====> Formatting code..."
 	@goimports -l -w -local ${PACKAGE_BASE} ${SRC}
 	@go run tools/pedantic_imports/main.go ${PACKAGE_BASE} ${SRC}
 
-install-golangci-lint:
-	@echo "> Installing golangci-lint..."
-	cd tools && $(GOCMD) install github.com/golangci/golangci-lint/cmd/golangci-lint
+## generate: Generate mocks
+generate: install-mockgen
+	@echo "=====> Generating mocks..."
+	$(GOCMD) generate ./...
 
+## lint: Check the code
 lint: install-golangci-lint
-	@echo "> Linting code..."
+	@echo "=====> Linting code..."
 	@golangci-lint run -c golangci.yaml
 
+## test: Run unit and acceptance tests
 test: unit acceptance
 
-# append coverage arguments
-ifeq ($(TEST_COVERAGE), 1)
-unit: GOTESTFLAGS:=$(GOTESTFLAGS) -coverprofile=./out/tests/coverage-unit.txt -covermode=atomic
-endif
-ifeq ($(NO_DOCKER),)
-unit: GOTESTFLAGS:=$(GOTESTFLAGS) --tags=example
-endif
```

Comment:
It looks like the removal of this is causing our tests to fail, because it's not correctly generating the unit test coverage files. I think we should put them back in